engine/crates/partial-caching/src/output/shapes/fragment_iter.rs

```diff
@@ -0,0 +1,83 @@
+use cynic_parser::executable::{FragmentDefinition, InlineFragment, Selection};
+
+use crate::{query_subset::FilteredSelectionSet, QuerySubset};
+
+use super::building::MergedSelection;
+
+/// An iterator over the fragments of a selection set.
+///
+/// This will recurse into any selection sets nested inside fragments, but not fields.
+pub struct FragmentIter<'doc, 'ctx> {
+    root_selection: std::slice::Iter<'ctx, MergedSelection<'doc>>,
+    iter_stack: Vec<FilteredSelectionSet<'doc, 'ctx>>,
+    subset: &'ctx QuerySubset,
```

Comment:
might just be me, not being in the middle of the code, but keeping the full name `query_subset` everywhere QuerySubset is used would be a bit clearer I think. `subset` when dealing with selection sets, sub-selection sets, etc... feels a bit ambiguous to me

Reply:
I think I'm going to leave this one as is, at least for now - I've used `subset` consistently throughout the entire crate.  There are 84 results when I search for the word, and I can't see a single one that's not referring to QuerySubset.

Can maybe rename it later if you continue to think it's worthwhile, but probably doesn't belong in this PR.